src/QueryBuilder/QueryBuilder.js

```diff
@@ -1 +1,17 @@
-export const QueryBuilder = () => 'QueryBuilder';
\ No newline at end of file
+import PropTypes from 'prop-types';
+import { ResultViewer } from './ResultViewer';
+import { QueryBuilderModal } from './QueryBuilderModal';
+
+const VIEWER = 'viewer';
+const BUILDER = 'builder';
+
+export const QueryBuilder = ({ componentType, ...rest }) => {
+  if (componentType === VIEWER) return <ResultViewer {...rest} />;
+  if (componentType === BUILDER) return <QueryBuilderModal {...rest} />;
```

Comment:
probably `QueryBuilder` is enough

Reply:
oops, just noticed that main component is called QueryBuilder :)

Reply:
Changed structure a little